tests/test_edgeql_expressions.py

```diff
@@ -2170,6 +2170,26 @@ async def test_edgeql_expr_valid_collection_16(self):
             [True]
         )
 
+    async def test_edgeql_expr_valid_minmax_01(self):
+        for val in get_test_values():
+            for fn in ['min', 'max']:
+                query = f"""SELECT {fn}({val}) = {val};"""
+                await self.assert_query_result(query, {True})
+
+    async def test_edgeql_expr_valid_minmax_02(self):
+        for val in get_test_values():
+            for fn in ['min', 'max']:
+                # same as the previous test, but for arrays
+                query = f"""SELECT {fn}([{val}]) = [{val}];"""
+                await self.assert_query_result(query, {True})
+
+    async def test_edgeql_expr_valid_minmax_03(self):
+        for val in get_test_values():
+            for fn in ['min', 'max']:
+                # same as the previous test, but for tuples
+                query = f"""SELECT {fn}(({val},)) = ({val},);"""
+                await self.assert_query_result(query, {True})
+
```

Comment:
What happens if we take `min` or `max` of an object type, especially one with nontrivial shape components?

I think it's worth adding a test for that, but if it doesn't work it's probably OK to disallow it for now. (Or for ever? It isn't particularly semantically meaningful.)

Reply:
Semantically the "value" of an object is its ID.  Shape components aren't considered to be part of that value (unless cast to `json`).  Even though there's no intrinsic external meaning to sorting objects by value, I don't think we should ban this case.  `ORDER BY` guarantees certain stable order of elements in set, which by itself might be useful in some cases and which is why we consider all types to be orderable.

Reply:
I've added tests for min/max on objects. They're ordered by their `id`, so min/max should respect that as well.